all: bump rtnetlink to 0.16.0 and fix compatibility issues

ffmancera opened the pull request:
I have separated the commits by the kind of error they are fixing. This should make the review easier.

ffmancera committed:
flags: update enum flags to plural form

netlink-packet-route has updated the enum flags name to the plural form.
In addition, they are bitflags now so we can iter without converting
them as slice.

The following enums changed:
LinkFlag -> LinkFlags
AddressFlag -> AddressFlags
RouteNextHopFlag -> RouteNextHopFlags
RouterFlag -> RouterFlags

Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

ffmancera committed:
vxlan: stop using parse_as_ipv4/6

As netlink-packet-route changed the vxlan group/6 and local/6 types from
raw array of u8 to Ipv4Addr and Ipv6Addr we don't need to use the helper
functions anymore.

Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

ffmancera committed:
query: use the newly introduced enums

netlink-packet-route introduced new enums for different properties. Use
them instead of the raw value.

The new enums are: BridgeVlanInfoFlags, BondMode, BondArpValidate,
IpVlanFlags, IpVlanMode, MacVlanMode.

Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

ffmancera committed:
iface: Use IfNetnsId instead of NetnsId LinkAttribute

netlink-packet-route has updated the LinkAttribute NetnsId property to
IfNetnsId.

ffmancera committed:
conf: Use LinkMessageBuilder and RouteMessageBuilder

rtnetlink introduced LinkMessageBuilder and RouteMessageBuilder in order
to send the messages.

Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

ffmancera committed:
bridge: no need to convert vlan_filtering from u8 to bool

netlink-packet-route has updated the Bridge nla vlan_filtering type from
u8 to bool. We don't need to convert it anymore and we can use it
directly.

Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

ffmancera committed:
dependency: Bump `netlink-packet-route` to 0.22.0

In order to do this and avoid duplicated dependencies, we need to bump
rtnetlink to 0.16.0 and libc to 0.2.155. Otherwise, it won't be possible
to compile it as rtnetlink bumped the Nix version which requires libc
0.2.155 or greater.

Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

cathay4t commented on src/lib/Cargo.toml:
```diff
@@ -19,15 +19,15 @@ crate-type = ["lib"]
-rtnetlink = "0.14.0"
-netlink-packet-route = "0.19.0"
+rtnetlink = "0.15.0"
+netlink-packet-route = "0.21.0"
```
The rtnetlink already reexport `netlink-packet-route`, please change code to use `rtnetlink::packet_route`, so we can remove this dependency.

ffmancera committed:
integ_tests: adjust integration tests

Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

ffmancera committed:
vrf: use InfoVrfPort instead of parsing the netlink attribute

InfoVrfPort was introduced in netlink-packet-route 0.20.1, use it
instead of parsing the netlink attribute from the buffer.

Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

ffmancera committed:
all: use rtnetlink `netlink-*` re-exports

As rtnetlink is re-exporting the needed `netlink-*` crates we don't need
to depend on them directly anymore.

Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

ffmancera committed:
routes: fix route filtering when querying

In route querying, when we hit `get_routes()` we always have a
NetStateRouteFilter but it all the fields are None. This was causing
nispor to set NETLINK_GET_STRICT_CHK always which is wrong.

On the other hand, when specifying a filtering value, we must make sure
the values we are not filtering by are set to the wildcard option.

ffmancera commented:
@cathay4t all passing now! could you do a minor release in netlink-packet-route containing the fixes and then another one in rtnetlink using it? Thanks!

cathay4t commented:
Please use rtnetlink 0.16.0.
